fabfile.py

```diff
@@ -176,13 +184,52 @@ def deploy(c, target=None, commit=None):
 
 
 @task
-def rollback(c, release=None):
+def rollback(c, target=DEFAULT_TARGET, release=None):
     with Connection(c.deploy.host, user=c.deploy.user, config=c.config) as c:
         setup(c, release=release)
         update(c)
         publish(c)
         finish(c)
 
+# Please add the "target" parameter if you are adding more @task functions
+#  to allow custom targets to be used (regardless if your function itself will use it or not)
+
+
+# Source: https://realpython.com/python-command-line-arguments/#a-few-methods-for-parsing-python-command-line-arguments
+def parseArgs(systemArgs, argStart=1):
+    opts = {"":[]}
+    cmdLineLength = len(systemArgs)
+    i = argStart
+
+    while i < cmdLineLength:
+        cmdArg = systemArgs[i]
+        if cmdArg.startswith("-"):
+            cmdArg2 = systemArgs[i + 1] if (i + 1) < cmdLineLength else ""
+            if cmdArg2.startswith("-"):
+                cmdArg2 = ""
+            else:
+                i += 1
+            # cmdArg2 = cmdArg.split("=", 1)
+            # cmdArg2 = cmdArg2[1] if len(cmdArg2) == 2 else ""
+            if cmdArg not in opts:
+                opts[cmdArg] = cmdArg2
+        else:
+            opts[""].append(cmdArg)
+        i += 1
+    
+    return opts
+
+def getTarget(options, default=DEFAULT_TARGET):
+    target = options.get(TARGET_FLAG, default)
+    if target not in configs:
+        message = "\n\tTarget Configuration at {} is not a valid entry".format(TARGET_FLAG)
```

Comment:
'\t'? Really?

Reply:
I have an "assert target in configs, message" towards the end of the "if" statement
It's just for styling such that an error message comes out like this:

AssertionError ... :
     Target Configuration at ...
     Invalid Entry: NotATarget

Reply:
Rather than one long line of confusion